src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -1162,6 +1162,34 @@ def _get_server_key_name_from_uri(uri):
     return '{}_{}_{}'.format(vault, key, version)
 
 
+#####
+#           sql server dns-alias
+#####
+
+
+def server_dns_alias_set(
+        client,
+        resource_group_name,
+        server_name,
+        dns_alias_name,
+        original_server_name,
+        original_subscription_id=None,
+        original_resource_group_name=None):
```

Comment:
Sorry to be a pain but can you also add logic to check if the alias already exists at the targeted location, and if so then do nothing? this will allow the command to be idempotent which is very helpful for scripts. And also add a test for this (yes, rerunning the test will be slow, sorry :) )

Reply:
I have added this case to tests. However, update command is idempotent already on ARM level, so imho there is no need to add additional logic here